src/Symfony/Component/Yaml/Dumper.php

```diff
@@ -39,13 +39,18 @@ public function __construct(int $indentation = 4)
     /**
      * Dumps a PHP value to YAML.
      *
-     * @param mixed                     $input  The PHP value
-     * @param int                       $inline The level where you switch to inline YAML
-     * @param int                       $indent The level of indentation (used internally)
-     * @param int-mask-of<Yaml::DUMP_*> $flags  A bit field of Yaml::DUMP_* constants to customize the dumped YAML string
+     * @param mixed                     $input        The PHP value
+     * @param int                       $inline       The level where you switch to inline YAML
+     * @param int                       $indent       The level of indentation (used internally)
+     * @param int-mask-of<Yaml::DUMP_*> $flags        A bit field of Yaml::DUMP_* constants to customize the dumped YAML string
+     * @param int                       $nestingLevel The level of nesting (used internally)
      */
-    public function dump(mixed $input, int $inline = 0, int $indent = 0, int $flags = 0): string
+    public function dump(mixed $input, int $inline = 0, int $indent = 0, int $flags = 0, int $nestingLevel = 0): string
```

Comment:
I would prefer to not expose the nesting level. We could instead move the current code of the `dump()` method into a new (`private`) `doDump()` method which gets this new argument, call it everywhere inside this class and pass `0` when being called from the new `dump()` method.

Reply:
I like it very much. Updated with your suggestion. Thanks!